File: programmers/네트워크.py
def solution(n, computers):
    answer = 0
    com = [0] * n
    
    def dfs(v):
        for i in range(n):
            if computers[v][i] == 1 and com[i] == 0: # 해당 노드와 연결된 것 확인
                com[i] = 1
                print(i,"와 연결돼있음")
                dfs(i)
        
    for j in range(n):
        if com[j] == 0:
            dfs(j)
            answer += 1
    
    return answer

File: programmers/네트워크.java
import java.util.*;
import java.io.*;

class Solution {
    
    boolean[] visited;
    int answer = 0;
    int[][] realcomputers;
    
    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        realcomputers = computers;
        
        for(int i=0; i<n; i++){
            if(!visited[i]) bfs(i,n);    
        }
        
        return answer;
    }
    
    void bfs(int start, int n){
        Queue<Integer> queue = new ArrayDeque<>();
        queue.offer(start);
        
        while(!queue.isEmpty()){
            int tmp = queue.poll();
            visited[tmp] = true;
            
            for (int i=0; i<n; i++){
                if (realcomputers[tmp][i] == 1 && !visited[i]) {
                    queue.offer(i);
                }
            }
            
        }
        answer++;
    }
    
    
}
